File: NetCore/ConsoleOop/ConsoleOop/Finders/IFileFinder.cs
using System.Collections;
using ConsoleOop.Configs;

namespace ConsoleOop.Finders
{
    /// <summary>
    /// 備份來源介面
    /// </summary>
    interface IFileFinder : IEnumerable, IEnumerator
    {
        /// <summary>
        /// 來源讀取
        /// </summary>
        /// <param name="config">檔案設定</param>
        void FileFinder(Config config);
    }
}

File: Laravel/app/Services/Finders/IFileFinder.php
<?php
/**
 * Created by PhpStorm.
 * User: tsao
 * Date: 2017/11/12
 * Time: 下午 07:35
 */

namespace App\Services\Finders;


use App\Services\Configs\Config;

/**
 * 備份來源介面
 * Interface IFileFinder
 * @author steve.tsao
 * @package App\Services\Finders
 */
interface IFileFinder
{
    /**
     * 來源讀取
     * @author steve.tsao
     * @param Config $config 檔案設定
     */
    function FileFinder(Config $config);
}
